src/components/Icon/Icon.tsx

```diff
@@ -130,10 +136,26 @@ class Icon extends UIComponent<Extendable<IIconProps>, any> {
     )
   }
 
+  isFontBased(themeIcons: ThemeIcons): boolean {
+    const { name } = this.props
```

Comment:
nit: from the function signature, it is not obvious that it checks current icon (in this).
I would change the signature to `isFontBased(name, themeIcons)`